qlty-check/src/executor.rs

```diff
@@ -59,6 +59,10 @@ impl Executor {
 
     pub fn install_and_invoke(&self) -> Result<Results> {
         let install_messages = self.install()?;
+        if self.plan.settings.install_only {
```

Comment:
I feel like rather than having the Executor reach _through_ the Plan up to the Settings that the Planner should be responsible for determining what to do an encoding that information into the Plan.

The Planner reads Settings (and other things like files) and turns it a Plan. The Executor runs the Plan.

Do we have other cases today where the executor reads from `self.settings.plan`?

Reply:
Yes there are a total of 6 other instances of `self.plan.settings` in the executor right now.

Reply:
OK, what I would suggest is that we take a look at what would be needed to address those.

The reason that I would do that before this is because this increases the number of `self.plan.settings` references when I think we need to drive that down to zero